splash/lua_modules/splash.lua

```diff
@@ -106,13 +106,13 @@ end
 -- error handling.
 --
 local Splash = {}
-Splash.__index = Splash
 
 local private = {}
 local PRIVATE_PREFIX = "private_"
 
 function Splash._create(py_splash)
-  local self = {args=py_splash.args}
+  local self = {args=py_splash.args, attr_to_settr_map={},
+                attr_to_gettr_map={}}
```

Comment:
What about making attr_to_getter_map and attr_to_setter_map private? Currently they will be suggested by the autocompletion engine if you use Splash-Jupyter, and they shouldn't. Adding `_` to the name may be enough to hide them from autocompleter - see [here](https://github.com/scrapinghub/splash/blob/fe80d031bf4d45e6f8b341f643a60018607fef9c/splash/kernel/completer.py#L94). This won't hide them from user, but I don't think it is a problem. An alternative is to put these maps outside Splash object (like it is done for `private`), or to protect against accessing them in `__newindex` / `__index`. The way private attributes are hidden is not perfect because it requires at most 1 Splash object per Lua runtime; we'll have to refactor this code if we start supporting multiple browser tabs in a single Lua script.

We should also rename these methods: `settr` vs `setter` saves only a single character :) Just `_setters` and `_getters` sounds better to me, but if you prefer longer names that's fine.

Could you please also indent it like dicts are indented in Splash Python code?

```lua
local self = {
  args=py_splash.args, 
  _setters={},
  _getters={},
}
```